### PR TITLE
require core if not loaded when requiring extensions

### PR DESCRIPTION
**Pull request by mojavelinux**
- force core to load if extensions are loaded first

### CODE REVIEW

**Comment by mojavelinux** [comment]
This is specifically for Jekyll, which loads core lazily. Therefore, extensions could get loaded before core, so we want to guarantee that order.

File: lib/asciidoctor/extensions.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'asciidoctor'.to_s unless defined? Asciidoctor
```
Comment (Mogztter) [inline lib/asciidoctor/extensions.rb]: @mojavelinux just for my curiosity, why are you using the `to_s` on a String ? As far as I know, `to_s` returns the String representation of an Object, so `'asciidoctor'` and `'asciidoctor'.to_s` should be equivalent no ?

**Comment by mojavelinux** [comment]
It's a trick to hide the import from Opal since it's never needed in that
environment.

**Comment by Mogztter** [comment]
Oh I see, thanks for explaining!

**Comment by jirutka** [comment]
This really should be documented using a comment, because no one else can know the purpose of this weird hack and even you will surely not remember it after few weeks. Just a common sense…

**Comment by mojavelinux** [comment]
Fair enough. I added a note above each usage. Opal only supports static require statements, which is why this works.

**Pull request merged**